File: Numeros Pseudo-Aleatorios/Add With Carry/AddWithCarry.dart
/*
 PRNG .......: Add With Carry (AWC)
 Ano ........: 1991
 Autor ......: George Marsaglia
*/

import 'dart:io';

int AWCNextRandom(int Xn) {
  Xn = (Xn + 12345) % 4294967296;
  return Xn;
}

void AWCInitialize(int seed, List<int> Xn) {
  Xn[0] = seed;
}

void main() {

  stdout.writeln("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)");
  stdout.writeln("PSEUDORANDOM NUMBER GENERATOR (PRNG)");
  stdout.writeln("PADRAO: Add With Carry");
  stdout.writeln("==========================================");
  stdout.writeln();
  stdout.writeln("    Semente: 1234        Semente: TIME");
  stdout.writeln("    -------------        -------------");

  List<int> Xn1 = List.filled(1, 0);
  List<int> Xn2 = List.filled(1, 0);
  int i;
  int seed1, seed2;

  seed1 = 1234;
  AWCInitialize(seed1, Xn1);

  seed2 = DateTime.now().millisecondsSinceEpoch;
  AWCInitialize(seed2, Xn2);

  for (i = 1; i <= 10; i++) {
    Xn1[0] = AWCNextRandom(Xn1[0]);
    Xn2[0] = AWCNextRandom(Xn2[0]);

    stdout.write(' ' * (17 - Xn1[0].toString().length) + '${Xn1[0]}');
    stdout.write('        ');
    stdout.write(' ' * (13 - Xn2[0].toString().length) + '${Xn2[0]}');
    stdout.writeln();
  }

}

File: Numeros Pseudo-Aleatorios/Add With Carry/AddWithCarry.py
"""
 PRNG .......: Add With Carry (AWC)
 Ano ........: 1991
 Autor ......: George Marsaglia
"""

import time

def AWCNextRandom(Xn):
    Xn = (Xn + 12345) % 4294967296
    return Xn

def AWCInitialize(seed):
    Xn = seed
    return Xn

if __name__ == '__main__':
    print("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)")
    print("PSEUDORANDOM NUMBER GENERATOR (PRNG)")
    print("PADRAO: Add With Carry")
    print("==========================================")
    print()
    print("    Semente: 1234        Semente: TIME")
    print("    -------------        -------------")

    i = 0
    seed1, seed2 = 1234, int(time.time())

    Xn1 = AWCInitialize(seed1)
    Xn2 = AWCInitialize(seed2)

    for i in range(1, 11):
        Xn1 = AWCNextRandom(Xn1)
        Xn2 = AWCNextRandom(Xn2)
        print(f"{Xn1:>17}        {Xn2:>13}")
